### PR TITLE
Add all positive cases for TomEE

### PR DESCRIPTION
**Pull request by jannahsherif**

### CODE REVIEW

File: .github/workflows/nightly.yml
[the remark this answers is not in the document]
Comment (jannahsherif) [reply .github/workflows/nightly.yml:1]: Ok will do

**Comment by jannahsherif** [comment]
Hello I just committed my new changes. After reviewing the feedback I add docker support and added the deleted cases

**Review by m-reza-rahman — CHANGES_REQUESTED** [review]
Please order by SE version (starting with the lowest), runtime/none (alphabetic, starting with none), EE version (starting with the lowest), profile (full, web, core), and Docker (no, yes).

So like this:

* Run Archetype for EE 8 Web Profile, SE 8, Payara
* Run Archetype for EE 8 Web Profile, SE 8, Payara, with Docker

Please order your part and the file overall - including existing cases. Confusing to read and evaluate otherwise for me.

**Comment by m-reza-rahman** [comment]
Please resolve conflicts with latest commits to master.

**Comment by jannahsherif** [comment]
Hello I tried implement the reviews and organizing the cases correctly

**Comment by m-reza-rahman** [comment]
At the bare minimum, the issue you need to resolve now are the conflicts before I can review again. Those will happen and are normal in a multi-developer project.

**Comment by jannahsherif** [comment]
hi i reviewed the conflicts

**Comment by m-reza-rahman** [comment]
OK. I will review if I have some time during the work week. Otherwise, may need to wait until the weekend.

**Review by m-reza-rahman — CHANGES_REQUESTED** [review]
Please double check ordering/duplication. It is still incorrect. Check the latest successful merges for reference if needed.

**Review by m-reza-rahman — CHANGES_REQUESTED** [review]
This PR should be focused on TomEE. Please remove changes related to WildFly.

**Comment by jannahsherif** [comment]
In my latest commit "minor fixes" i only added the cases for tomee i did not change anything else in the original nighly

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -9,23 +9,23 @@ jobs:
   build-archetype:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:12]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -9,23 +9,23 @@ jobs:
...
       - name: Set up Java SE 8
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:14]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -9,23 +9,23 @@ jobs:
...
         with:
           distribution: "temurin"
           java-version: 8
 
       - name: Cache Maven packages
-        uses: actions/cache@v3
+        uses: actions/cache@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:20]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 4]
```diff
@@ -9,23 +9,23 @@ jobs:
...
         with:
           path: ~/.m2
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
 
       - name: Build Archetype
         run: mvn clean install --file archetype/pom.xml
-      
+
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:28]: Why is this white space change needed.

File: .github/workflows/nightly.yml [hunk 5]
```diff
@@ -125,11 +125,16 @@ jobs:
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=9.1 -Dprofile=web -DjavaVersion=8 -Druntime="open-liberty" -Ddocker=yes -DoutputDirectory="app/open-liberty" -Dgoals="clean package"
           rm -rf app/open-liberty
-      
+
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:128]: Why is this white space change needed?

File: .github/workflows/nightly.yml [hunk 6]
```diff
@@ -139,6 +144,8 @@ jobs:
       - name: Run Archetype for EE 8, SE 8, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=full -DjavaVersion=8 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:147]: Please remove this. This is not related to TomEE.

File: .github/workflows/nightly.yml [hunk 7]
```diff
@@ -149,10 +156,12 @@ jobs:
       - name: Run Archetype for EE 8 Web Profile, SE 8, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=web -DjavaVersion=8 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:159]: Please remove this. This is not related to TomEE.

File: .github/workflows/nightly.yml [hunk 8]
```diff
@@ -149,10 +156,12 @@ jobs:
...
+          docker rmi test-image
           rm -rf app/wildfly
 
       - name: Set up Java SE 11
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:164]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 9]
```diff
@@ -291,17 +300,32 @@ jobs:
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=core -DjavaVersion=11 -Druntime="open-liberty" -Ddocker=yes -DoutputDirectory="app/open-liberty" -Dgoals="clean package"
           rm -rf app/open-liberty
-      
+
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:303]: Why is this white space change needed?

File: .github/workflows/nightly.yml [hunk 10]
```diff
@@ -310,6 +334,8 @@ jobs:
       - name: Run Archetype for EE 8, SE 11, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=full -DjavaVersion=11 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:337]: Please remove this. This is not related to TomEE.

File: .github/workflows/nightly.yml [hunk 11]
```diff
@@ -320,6 +346,8 @@ jobs:
       - name: Run Archetype for EE 8 Web Profile, SE 11, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=web -DjavaVersion=11 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:349]: Please remove this. This is not related to TomEE.

File: .github/workflows/nightly.yml [hunk 12]
```diff
@@ -330,6 +358,8 @@ jobs:
       - name: Run Archetype for EE 10, SE 11, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=full -DjavaVersion=11 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:361]: Please remove this. This is not related to TomEE.

File: .github/workflows/nightly.yml [hunk 13]
```diff
@@ -340,6 +370,8 @@ jobs:
       - name: Run Archetype for EE 10 Web Profile, SE 11, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=web -DjavaVersion=11 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:373]: Please remove this. This is not related to TomEE.

File: .github/workflows/nightly.yml [hunk 14]
```diff
@@ -350,10 +382,12 @@ jobs:
       - name: Run Archetype for EE 10 Core Profile, SE 11, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=core -DjavaVersion=11 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:385]: Please remove this. This is not related to TomEE.

File: .github/workflows/nightly.yml [hunk 15]
```diff
@@ -350,10 +382,12 @@ jobs:
...
+          docker rmi test-image
           rm -rf app/wildfly
 
       - name: Set up Java SE 17
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:390]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 16]
```diff
@@ -497,12 +531,32 @@ jobs:
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=core -DjavaVersion=17 -Druntime="open-liberty" -Ddocker=yes -DoutputDirectory="app/open-liberty" -Dgoals="clean package"
           rm -rf app/open-liberty
-  
+
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:534]: Why is this white space change needed?

File: .github/workflows/nightly.yml [hunk 17]
```diff
@@ -556,21 +618,22 @@ jobs:
...
       - name: Setup Java
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           distribution: "temurin"
           java-version: 11
 
       - name: Cache Maven packages
-        uses: actions/cache@v3
+        uses: actions/cache@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:636]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 18]
```diff
@@ -556,21 +618,22 @@ jobs:
...
 
-
   build-ui:
     runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: Setup Java
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:630]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 19]
```diff
@@ -556,21 +618,22 @@ jobs:
...
+          docker build -t test-image app/wildfly/jakartaee-hello-world
+          docker rmi test-image
           rm -rf app/wildfly
 
-
   build-ui:
     runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:628]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 20]
```diff
@@ -516,6 +570,8 @@ jobs:
       - name: Run Archetype for EE 8, SE 17, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=full -DjavaVersion=17 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:573]: Please remove this. This is not related to TomEE.

File: .github/workflows/nightly.yml [hunk 21]
```diff
@@ -526,6 +582,8 @@ jobs:
       - name: Run Archetype for EE 8 Web Profile, SE 17, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=web -DjavaVersion=17 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:585]: Please remove this. This is not related to TomEE.

File: .github/workflows/nightly.yml [hunk 22]
```diff
@@ -536,6 +594,8 @@ jobs:
       - name: Run Archetype for EE 10, SE 17, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=full -DjavaVersion=17 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:597]: Please remove this. This is not related to TomEE.

File: .github/workflows/nightly.yml [hunk 23]
```diff
@@ -546,6 +606,8 @@ jobs:
       - name: Run Archetype for EE 10 Web Profile, SE 17, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=web -DjavaVersion=17 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:609]: Please remove this. This is not related to TomEE.

File: .github/workflows/nightly.yml [hunk 24]
```diff
@@ -556,21 +618,22 @@ jobs:
       - name: Run Archetype for EE 10 Core Profile, SE 17, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=core -DjavaVersion=17 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:621]: Please remove this. This is not related to TomEE.

**Review by m-reza-rahman — CHANGES_REQUESTED** [review]
I have now added specific in-place feedback for you. Please address them. Then I can review hopefully one last time before merge.

**Comment by jannahsherif** [comment]
Thank you for the detailed explanation! I resolved all the conflicts but I am running into a problem with the space changes. If i remove the extra space it highlights it in red. If I readd it, it shows one line red then the one after green. How can i get it not be red or green?

**Comment by m-reza-rahman** [comment]
It looks honestly like you have some sync issues. What I suggest is closing this PR, saving aside your changes, forking fresh again, re-apply your changes, and then open a new PR. That might be easiest at this point.

**Pull request closed without merging**